workers.tf

```diff
@@ -6,7 +6,7 @@ resource "aws_autoscaling_group" "workers" {
     "-",
     compact(
       [
-        aws_eks_cluster.this[0].name,
+        var.cluster_name,
```

Comment:
Please use the splat syntax instead : `coalescelist(aws_eks_cluster.this[*].name, [""])[0]` This is true for all your change.